Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -702,6 +706,12 @@ static AccessibilityPreferences accessibilityPreferences()
 #if HAVE(MEDIA_ACCESSIBILITY_FRAMEWORK)
     addCFNotificationObserver(mediaAccessibilityPreferencesChangedCallback, kMAXCaptionAppearanceSettingsChangedNotification);
 #endif
+#if HAVE(POWERLOG_TASK_MODE_QUERY) && ENABLE(GPU_PROCESS)
+    m_powerLogObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kPLTaskingStartNotification object:nil queue:[NSOperationQueue currentQueue] usingBlock:^(NSNotification *notification) {
```

Comment:
Should we use mainQueue instead of currentQueue here? The documentation suggests that currentQueue can return nil, which would have the effect of running our callback on whatever thread posts the notification, which could cause WebKit heap corruption in extremely rare circumstances, which are my least favorite circumstances. :P

Reply:
> Should we use mainQueue instead of currentQueue here? The documentation suggests that currentQueue can return nil, which would have the effect of running our callback on whatever thread posts the notification, which could cause WebKit heap corruption in extremely rare circumstances, which are my least favorite circumstances. :P

That is a good catch. Changed to mainQueue in latest patch.